src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclets/toolkit/util/Utils.java

```diff
@@ -2924,4 +2947,237 @@ public String toString() {
             return first + ":" + second;
         }
     }
+
+    /**
+     * Return the set of preview language features used to declare the given element.
+     *
+     * @param e the Element to check.
+     * @return the set of preview language features used to declare the given element
+     */
+    @SuppressWarnings("preview")
+    public Set<DeclarationPreviewLanguageFeatures> previewLanguageFeaturesUsed(Element e) {
+        Set<DeclarationPreviewLanguageFeatures> result = new HashSet<>();
+
+        if (e.getKind() == ElementKind.RECORD) {
+            result.add(DeclarationPreviewLanguageFeatures.RECORD);
+        }
+
+        if ((e.getKind().isClass() || e.getKind().isInterface()) &&
+            e.getModifiers().contains(Modifier.SEALED)) {
+            List<? extends TypeMirror> permits = ((TypeElement) e).getPermittedSubclasses();
+            boolean hasLinkablePermits = permits.stream()
+                                                .anyMatch(t -> isLinkable(asTypeElement(t)));
+            if (hasLinkablePermits) {
+                result.add(DeclarationPreviewLanguageFeatures.SEALED_PERMITS);
+            } else {
+                result.add(DeclarationPreviewLanguageFeatures.SEALED);
+            }
+        }
+
+        return result;
+    }
+
+    public enum DeclarationPreviewLanguageFeatures {
+
+        SEALED(List.of("sealed")),
+        SEALED_PERMITS(List.of("sealed", "permits")),
+        RECORD(List.of("record"));
```

Comment:
I'm guessing this is about to go away soon?

Reply:
Right, this is likely to go away soon.